ydb/core/kqp/opt/logical/kqp_opt_log_indexes.cpp

```diff
@@ -334,15 +399,146 @@ TExprBase DoRewriteIndexRead(const TReadMatch& read, TExprContext& ctx,
     }
 }
 
+TExprBase DoRewriteTopSortOverKMeansTree(
+    const TReadMatch& read, const TMaybeNode<TCoFlatMap>& flatMap, const TExprNode& lambdaArgs, const TExprBase& lambdaBody, const TCoTopBase& top,
+    TExprContext& ctx, const TKqpOptimizeContext& kqpCtx,
+    const TKikimrTableDescription& tableDesc, const TIndexDescription& indexDesc, const TKikimrTableMetadata& implTable)
+{
+    Y_ASSERT(indexDesc.Type == TIndexDescription::EType::GlobalSyncVectorKMeansTree);
+    const auto* levelTableDesc = &kqpCtx.Tables->ExistingTable(kqpCtx.Cluster, implTable.Name);
+    const auto* postingTableDesc = &kqpCtx.Tables->ExistingTable(kqpCtx.Cluster, implTable.Next->Name);
+    Y_ASSERT(!implTable.Next->Next);
```

Comment:
ensure? 

.Next->Next what is that? 

am i correctly understand, that Next points to the next impl table over the of index? 
and here we check, that we have at most 2 tables for vector index?

Reply:
Yes, right